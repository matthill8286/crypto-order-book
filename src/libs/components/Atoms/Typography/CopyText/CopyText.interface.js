
//# sourceMappingURL=CopyText.interface.js.map