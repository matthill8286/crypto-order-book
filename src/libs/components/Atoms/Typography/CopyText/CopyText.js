function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { Typo } from "../Typo";

var getMargins = function getMargins(fontSize) {
  if (fontSize && (fontSize === 'xs' || fontSize === 'sm')) {
    return 'xs 0 md';
  } else {
    return '';
  }
};

export var CopyText = function CopyText(_ref) {
  var _ref$tag = _ref.tag,
      tag = _ref$tag === void 0 ? 'p' : _ref$tag,
      _ref$withMargins = _ref.withMargins,
      withMargins = _ref$withMargins === void 0 ? false : _ref$withMargins,
      _ref$bold = _ref.bold,
      bold = _ref$bold === void 0 ? false : _ref$bold,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'grey5' : _ref$color,
      _ref$fontSize = _ref.fontSize,
      fontSize = _ref$fontSize === void 0 ? 'xs' : _ref$fontSize,
      children = _ref.children,
      margin = _ref.margin,
      restProps = _objectWithoutProperties(_ref, ["tag", "withMargins", "bold", "color", "fontSize", "children", "margin"]);

  return React.createElement(Typo, _extends({
    tag: tag,
    color: color,
    fontSize: fontSize,
    weight: bold ? 'bold' : 'regular',
    margin: withMargins && fontSize && (typeof fontSize === 'string' ? getMargins(fontSize) : getMargins(fontSize === null || fontSize === void 0 ? void 0 : fontSize.mobile)) || margin
  }, restProps), children);
};
CopyText.displayName = 'CopyText';
//# sourceMappingURL=CopyText.js.map