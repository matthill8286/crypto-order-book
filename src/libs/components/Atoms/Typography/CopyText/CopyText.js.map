{"version":3,"sources":["../../../../../src/components/Atoms/Typography/CopyText/CopyText.tsx"],"names":["React","Typo","getMargins","fontSize","CopyText","tag","withMargins","bold","color","children","margin","restProps","mobile","displayName"],"mappings":";;;;;;AAAA,OAAOA,KAAP,MAAkB,OAAlB;AAEA,SAASC,IAAT;;AAIA,IAAMC,UAAU,GAAG,SAAbA,UAAa,CAACC,QAAD,EAA8C;AAC/D,MAAIA,QAAQ,KAAKA,QAAQ,KAAK,IAAb,IAAqBA,QAAQ,KAAK,IAAvC,CAAZ,EAA0D;AACxD,WAAO,SAAP;AACD,GAFD,MAEO;AACL,WAAO,EAAP;AACD;AACF,CAND;;AAQA,OAAO,IAAMC,QAAiC,GAAG,SAApCA,QAAoC;AAAA,sBAC/CC,GAD+C;AAAA,MAC/CA,GAD+C,yBACzC,GADyC;AAAA,8BAE/CC,WAF+C;AAAA,MAE/CA,WAF+C,iCAEjC,KAFiC;AAAA,uBAG/CC,IAH+C;AAAA,MAG/CA,IAH+C,0BAGxC,KAHwC;AAAA,wBAI/CC,KAJ+C;AAAA,MAI/CA,KAJ+C,2BAIvC,OAJuC;AAAA,2BAK/CL,QAL+C;AAAA,MAK/CA,QAL+C,8BAKpC,IALoC;AAAA,MAM/CM,QAN+C,QAM/CA,QAN+C;AAAA,MAO/CC,MAP+C,QAO/CA,MAP+C;AAAA,MAQ5CC,SAR4C;;AAAA,SAU/C,oBAAC,IAAD;AACE,IAAA,GAAG,EAAEN,GADP;AAEE,IAAA,KAAK,EAAEG,KAFT;AAGE,IAAA,QAAQ,EAAEL,QAHZ;AAIE,IAAA,MAAM,EAAEI,IAAI,GAAG,MAAH,GAAY,SAJ1B;AAKE,IAAA,MAAM,EACHD,WAAW,IACVH,QADD,KAEE,OAAOA,QAAP,KAAoB,QAApB,GACGD,UAAU,CAACC,QAAD,CADb,GAEGD,UAAU,CAAEC,QAAF,aAAEA,QAAF,uBAAEA,QAAD,CAA2BS,MAA5B,CAJf,CAAD,IAKAF;AAXJ,KAaMC,SAbN,GAcGF,QAdH,CAV+C;AAAA,CAA1C;AA4BPL,QAAQ,CAACS,WAAT,GAAuB,UAAvB","sourcesContent":["import React from 'react'\nimport { BoxDimensions, ThemeFontSizes } from '@/types/theme'\nimport { Typo } from '../Typo'\nimport { FontSizeMap } from '../Typo/Typo.interface'\nimport { CopyTextProps } from './CopyText.interface'\n\nconst getMargins = (fontSize?: ThemeFontSizes): BoxDimensions => {\n  if (fontSize && (fontSize === 'xs' || fontSize === 'sm')) {\n    return 'xs 0 md'\n  } else {\n    return ''\n  }\n}\n\nexport const CopyText: React.FC<CopyTextProps> = ({\n  tag = 'p',\n  withMargins = false,\n  bold = false,\n  color = 'grey5',\n  fontSize = 'xs',\n  children,\n  margin,\n  ...restProps\n}) => (\n  <Typo\n    tag={tag}\n    color={color}\n    fontSize={fontSize}\n    weight={bold ? 'bold' : 'regular'}\n    margin={\n      (withMargins &&\n        fontSize &&\n        (typeof fontSize === 'string'\n          ? getMargins(fontSize)\n          : getMargins((fontSize as FontSizeMap)?.mobile))) ||\n      margin\n    }\n    {...restProps}>\n    {children}\n  </Typo>\n)\n\nCopyText.displayName = 'CopyText'\n"],"file":"CopyText.js"}