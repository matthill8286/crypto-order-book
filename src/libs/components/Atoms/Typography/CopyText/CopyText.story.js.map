{"version":3,"sources":["../../../../../src/components/Atoms/Typography/CopyText/CopyText.story.tsx"],"names":["boolean","number","select","text","storiesOf","React","saiyanTheme","CopyText","module","add","sampleText","fontSizeOptions","fontSize","tagOptions","tagName","withMargins","multiple","showColors","selectedColor","Object","keys","color","isBold","limitLines"],"mappings":";;AAAA,SAASA,OAAT,EAAkBC,MAAlB,EAA0BC,MAA1B,EAAkCC,IAAlC,QAA8C,wBAA9C;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,WAAT;AAEA,SAASC,QAAT;AAEAH,SAAS,CAAC,gCAAD,EAAmCI,MAAnC,CAAT,CAAoDC,GAApD,CAAwD,UAAxD,EAAoE,YAAM;AACxE,MAAMC,UAAU,GAAGP,IAAI,CACrB,aADqB,oOAAvB;AAKA,MAAMQ,eAAe,GAAG,CAAC,KAAD,EAAQ,IAAR,EAAc,IAAd,EAAoB,IAApB,CAAxB;AACA,MAAMC,QAAQ,GAAGV,MAAM,CAAC,eAAD,EAAkBS,eAAlB,EAAmC,IAAnC,CAAvB;AACA,MAAME,UAA2C,GAAG,CAAC,GAAD,EAAM,MAAN,EAAc,KAAd,CAApD;AACA,MAAMC,OAAO,GAAGZ,MAAM,CAAC,UAAD,EAAaW,UAAb,EAAyB,GAAzB,CAAtB;AACA,MAAME,WAAW,GAAGf,OAAO,CAAC,cAAD,EAAiB,KAAjB,CAA3B;AACA,MAAMgB,QAAQ,GAAGhB,OAAO,CAAC,qBAAD,EAAwB,KAAxB,CAAxB;AACA,MAAMiB,UAAU,GAAGjB,OAAO,CAAC,qBAAD,EAAwB,KAAxB,CAA1B;AACA,MAAMkB,aAAa,GAAGhB,MAAM,CAAC,OAAD,EAAUiB,MAAM,CAACC,IAAP,CAAYd,WAAW,CAACe,KAAxB,CAAV,EAA0C,OAA1C,CAA5B;AACA,MAAMC,MAAM,GAAGtB,OAAO,CAAC,MAAD,EAAS,KAAT,CAAtB;AACA,MAAMuB,UAAU,GAAGtB,MAAM,CAAC,aAAD,EAAgB,CAAhB,CAAzB;AAEA,SACE,0CACE,oBAAC,QAAD;AACE,IAAA,GAAG,EAAEa,OADP;AAEE,IAAA,QAAQ,EAAEF,QAFZ;AAGE,IAAA,WAAW,EAAEG,WAHf;AAIE,IAAA,IAAI,EAAEO;AAJR,KAKOL,UAAU,IAAI;AAAEI,IAAAA,KAAK,EAAEH;AAAT,GALrB;AAME,IAAA,UAAU,EAAEK;AANd,MAOGb,UAPH,CADF,EAUGM,QAAQ,IACP,oBAAC,QAAD;AACE,IAAA,GAAG,EAAEF,OADP;AAEE,IAAA,QAAQ,EAAEF,QAFZ;AAGE,IAAA,WAAW,EAAEG,WAHf;AAIE,IAAA,IAAI,EAAEO;AAJR,KAKOL,UAAU,IAAI;AAAEI,IAAAA,KAAK,EAAEH;AAAT,GALrB;AAME,IAAA,UAAU,EAAEK;AANd,MAOGb,UAPH,CAXJ,EAqBGM,QAAQ,IACP,oBAAC,QAAD;AACE,IAAA,GAAG,EAAEF,OADP;AAEE,IAAA,QAAQ,EAAEF,QAFZ;AAGE,IAAA,WAAW,EAAEG,WAHf;AAIE,IAAA,IAAI,EAAEO;AAJR,KAKOL,UAAU,IAAI;AAAEI,IAAAA,KAAK,EAAEH;AAAT,GALrB;AAME,IAAA,UAAU,EAAEK;AANd,MAOGb,UAPH,CAtBJ,CADF;AAmCD,CApDD","sourcesContent":["import { boolean, number, select, text } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { saiyanTheme } from '@/styles/sc-vars-saiyan'\nimport { ThemeColors, ThemeFontSizes } from '@/types/theme'\nimport { CopyText } from './CopyText'\n\nstoriesOf('Design System/Atoms/Typography', module).add('CopyText', () => {\n  const sampleText = text(\n    'Sample text',\n    `Lorem ipsum dolorsit amet consectetur adipisicing elit. Lorem ipsum dolorsit amet consectetur adipisicing elit. Lorem ipsum dolorsit amet consectetur adipisicing elit. Lorem ipsum dolorsit amet consectetur adipisicing elit.`\n  )\n\n  const fontSizeOptions = ['xxs', 'xs', 'sm', 'md']\n  const fontSize = select('Type fontSize', fontSizeOptions, 'xs') as ThemeFontSizes\n  const tagOptions: (keyof JSX.IntrinsicElements)[] = ['p', 'span', 'div']\n  const tagName = select('Tag name', tagOptions, 'p')\n  const withMargins = boolean('With margins', false)\n  const multiple = boolean('Multiple paragraphs', false)\n  const showColors = boolean('Enable Custom Color', false)\n  const selectedColor = select('Color', Object.keys(saiyanTheme.color), 'black') as ThemeColors\n  const isBold = boolean('Bold', false)\n  const limitLines = number('Limit Lines', 0)\n\n  return (\n    <>\n      <CopyText\n        tag={tagName}\n        fontSize={fontSize}\n        withMargins={withMargins}\n        bold={isBold}\n        {...(showColors && { color: selectedColor })}\n        limitLines={limitLines}>\n        {sampleText}\n      </CopyText>\n      {multiple && (\n        <CopyText\n          tag={tagName}\n          fontSize={fontSize}\n          withMargins={withMargins}\n          bold={isBold}\n          {...(showColors && { color: selectedColor })}\n          limitLines={limitLines}>\n          {sampleText}\n        </CopyText>\n      )}\n      {multiple && (\n        <CopyText\n          tag={tagName}\n          fontSize={fontSize}\n          withMargins={withMargins}\n          bold={isBold}\n          {...(showColors && { color: selectedColor })}\n          limitLines={limitLines}>\n          {sampleText}\n        </CopyText>\n      )}\n    </>\n  )\n})\n"],"file":"CopyText.story.js"}