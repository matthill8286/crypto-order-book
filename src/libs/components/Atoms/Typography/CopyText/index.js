export * from "./CopyText";
//# sourceMappingURL=index.js.map