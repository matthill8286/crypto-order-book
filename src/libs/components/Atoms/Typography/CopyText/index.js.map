{"version":3,"sources":["../../../../../src/components/Atoms/Typography/CopyText/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './CopyText'\n"],"file":"index.js"}