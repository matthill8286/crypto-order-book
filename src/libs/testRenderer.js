import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map