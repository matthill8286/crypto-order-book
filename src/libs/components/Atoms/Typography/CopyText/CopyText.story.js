function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { boolean, number, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { saiyanTheme } from "../../../../styles/sc-vars-saiyan";
import { CopyText } from "./CopyText";
storiesOf('Design System/Atoms/Typography', module).add('CopyText', function () {
  var sampleText = text('Sample text', "Lorem ipsum dolorsit amet consectetur adipisicing elit. Lorem ipsum dolorsit amet consectetur adipisicing elit. Lorem ipsum dolorsit amet consectetur adipisicing elit. Lorem ipsum dolorsit amet consectetur adipisicing elit.");
  var fontSizeOptions = ['xxs', 'xs', 'sm', 'md'];
  var fontSize = select('Type fontSize', fontSizeOptions, 'xs');
  var tagOptions = ['p', 'span', 'div'];
  var tagName = select('Tag name', tagOptions, 'p');
  var withMargins = boolean('With margins', false);
  var multiple = boolean('Multiple paragraphs', false);
  var showColors = boolean('Enable Custom Color', false);
  var selectedColor = select('Color', Object.keys(saiyanTheme.color), 'black');
  var isBold = boolean('Bold', false);
  var limitLines = number('Limit Lines', 0);
  return React.createElement(React.Fragment, null, React.createElement(CopyText, _extends({
    tag: tagName,
    fontSize: fontSize,
    withMargins: withMargins,
    bold: isBold
  }, showColors && {
    color: selectedColor
  }, {
    limitLines: limitLines
  }), sampleText), multiple && React.createElement(CopyText, _extends({
    tag: tagName,
    fontSize: fontSize,
    withMargins: withMargins,
    bold: isBold
  }, showColors && {
    color: selectedColor
  }, {
    limitLines: limitLines
  }), sampleText), multiple && React.createElement(CopyText, _extends({
    tag: tagName,
    fontSize: fontSize,
    withMargins: withMargins,
    bold: isBold
  }, showColors && {
    color: selectedColor
  }, {
    limitLines: limitLines
  }), sampleText));
});
//# sourceMappingURL=CopyText.story.js.map