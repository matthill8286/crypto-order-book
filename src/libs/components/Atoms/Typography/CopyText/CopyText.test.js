import React from 'react';
import { renderWithTheme } from "../../../../testRenderer";
import { CopyText } from "./CopyText";
describe('CopyText', function () {
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(CopyText, null, "Lorem Ipsum"));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=CopyText.test.js.map