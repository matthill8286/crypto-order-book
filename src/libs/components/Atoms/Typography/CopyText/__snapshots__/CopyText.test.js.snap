// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyText renders correctly 1`] = `
<p
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 kjuzBv"
  color="grey5"
  fontSize="xs"
>
  Lorem Ipsum
</p>
`;
